### PR TITLE
asset/manifests: redact the pull-secret from the cluster object of the install-config

### PR DESCRIPTION
**Pull request by abhinavdahiya**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1686119
install-config is stores as a config map in the cluster, but the pull-secret in the install-config is a secret that
shouldn't be stores as config map.

The pull-secret is already available to cluster as secret `kube-system/coreos-pull-secret`.

### CODE REVIEW

File: pkg/asset/manifests/operators.go [hunk 1]
```diff
@@ -264,6 +269,11 @@ func (m *Manifests) Load(f asset.FileFetcher) (bool, error) {
 	return true, nil
 }
 
+func redactedInstallConfig(config types.InstallConfig) ([]byte, error) {
+	config.PullSecret = "XXX"
```
Comment (wking) [inline pkg/asset/manifests/operators.go]: nit: Can we just set it to the empty string?

**Comment by wking** [comment]
/lgtm
/hold

(I'll drop a new LGTM if you want to address [this][1] ;)  We can pull the hold once openshift/cluster-monitoring-operator#286 lands.

[1]: https://github.com/openshift/installer/pull/1379#discussion_r263195175

File: pkg/asset/manifests/operators.go [hunk 2]
```diff
@@ -91,9 +92,13 @@ func (m *Manifests) Generate(dependencies asset.Parents) error {
 	installConfig := &installconfig.InstallConfig{}
 	dependencies.Get(installConfig, ingress, dns, network, infra)
 
+	readactedConfig, err := redactedInstallConfig(*installConfig.Config)
```
Comment (staebler) [inline pkg/asset/manifests/operators.go]: s/readacted/redacted/

**Comment by wking** [comment]
/retest

Monitoring PR landed.  I'm not pulling the `/hold` because of the typo @staebler found, but I want to see if the monitoring failure was masking other consumers.

File: pkg/asset/manifests/operators.go (hunk 1 above)
Comment (abhinavdahiya) [reply pkg/asset/manifests/operators.go]: I wasn't sure if empty string would trigger some skip mechanism. so I purposefully chose to do `XXX`

but if you think empty is fine, done with https://github.com/openshift/installer/compare/e41fed9f996ec1cc65b05d9fb456ee78d7aca271..141cd3ce50caff295e6dcbad6976ee5e2f344846

**Comment by wking** [comment]
/lgtm

[e2e-aws][1]:

```
Flaky tests:

[sig-network] Services should be able to switch session affinity for service with type clusterIP [Suite:openshift/conformance/parallel] [Suite:k8s]

Failing tests:

[Feature:Prometheus][Conformance] Prometheus when installed on the cluster should report telemetry if a cloud.openshift.com token is present [Suite:openshift/conformance/parallel/minimal]
```

/retest

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1379/pull-ci-openshift-installer-master-e2e-aws/4418/build-log.txt

**Comment by abhinavdahiya** [comment]
Needs https://github.com/openshift/origin/pull/22271 to pass CI.

**Comment by abhinavdahiya** [comment]
https://github.com/openshift/origin/pull/22271 merged [30 mins ago](https://github.com/openshift/origin/pull/22271#event-2191449601)

/retest

**Comment by wking** [comment]
All green.

/hold cancel

**Pull request merged**